snuba/datasets/discover.py

```diff
@@ -55,6 +61,67 @@
 logger = logging.getLogger(__name__)
 
 
+EVENT_CONDITION = FunctionCallMatch(
+    None,
+    Param("function", Or([StringMatch(op) for op in BINARY_OPERATORS])),
+    (
+        ColumnMatch(None, None, StringMatch("type")),
+        Param("event_type", AnyExpression()),
+    ),
+)
+
+
+def match_query_to_table(
+    query: Query, events_only_columns: ColumnSet, transactions_only_columns: ColumnSet
+) -> (str, bool, bool):
+    condition = query.get_condition_from_ast()
+    condition_columns = set()
+    event_types = []
+    if condition:
+        for cond in condition:
+            if Any(Column).match(cond):
+                condition_columns.add(cond)
+
+            result = EVENT_CONDITION.match(cond)
+            if not result:
+                continue
+
+            event_type_param = result.expression("event_type")
+            event_type = None
+            if isinstance(event_type_param, Column):
+                event_type = result.expression("event_type").column_name
```

Comment:
I don't think this type checks. mypy does not know `result.expression(a string)` is a column. Also you can reuse the event_type_param above.